README.md

```diff
@@ -1,2 +1,106 @@
+
+### Set up an S3 storage
+
+We will need a storage for our fake data, for metaflow flow code packages, and for Argo artifacts (more below). You can use `argo-minio.yaml` [in the manifests folder](manifests/argo-minio.yaml) or set-up your own [via the MinIO documentation (for prototyping)](https://min.io/docs/minio/kubernetes/upstream/index.html)
+
+Create a secret for allowing other services to access MinIO and its bucket storage: 
+```
+kubectl create secret generic argo-artifacts
+--from-literal=accesskey=XXXXXXX
+--from-literal=secretkey=XXXXXXXXXXXXXXX
+-n argo
+```
+
+### Set up Argo for automation
+
+Installation: 
+1. [This installation was used](https://argoproj.github.io/argo-workflows/quick-start/): `kubectl apply -n argo -f https://github.com/argoproj/argo-workflows/releases/download/v3.4.8/install.yaml`
+2. Argo deployment is launched with: `kubectl -n argo port-forward deployment/argo-server 2746:2746` 
+
+Artifact Repository: 
+Argo needs an artifact repository (storage for flows to run). Here we will use MinIO that we previously installed. You can use the `artifact-repositories.yaml` [in the manifests folder](manifests/artifact-repositories.yaml)
+(The secret for minio we created before comes in here). You can set it up with: `kubectl apply -f artifact-repositories.yaml`
+
+Give Argo admin roles on cluster to access MinIO: : `kubectl create rolebinding argo-default-admin --clusterrole=admin --serviceaccount=argo:default --namespace=argo` (we also had to repete this rolebinding creating for argo and argo-server service accounts in the argo namespace. This rolebinding may have to be revised in a production environment where an admin role could be problematic.
+
+### Set up an IDE (optional)
+
+If for some reason, you need/want an IDE deployed (if for example you are working on a server). Install via Helm [a vscode service](https://artifacthub.io/packages/helm/inseefrlab/vscode) then access by port-forwarding `kubectl -n argo port-forward deployment/my-vscode 35547:8080`. You will probably have to give admin rights to this VSCode (service account: `my-vscode` over MinIO (as done for Argo)). Finally, you will also have to add Metaflow configuration variables to the ConfigMap of VSCode, for putting metaflow code packages in MinIO. 
+
+![configmap-my-vscode-configmaps3metaflow](https://github.com/SDSC-ORD/zarr-linked-data/assets/22447169/498a1306-5289-4527-9cfa-22e947d555e3)
+
+Then you just install the requirements with `pip install -r requirements.txt` and you're set to go! 
+
+# C. Example/Usage
+
+## Fake Zarr Data
+You may not have data under the zarr format yet. You can define some metadata instance objects (such as in `zarr_linked_data/data/original_data.jsonld`), then define your hierarchy levels, data level and other parameters for the creation of a random Zarr test store using `zarr_linked_data/fake_data_flow.py`.
```

Comment:
**comment(non-blocking):** This (versioned) data makes the repo quite large (350MB), which can be cumbersome / slow things down. Perhaps we could later consider moving it to a data repository (zenodo?). It would also require rewriting the git-history to entirely remove the files from the index, though (with git commands or [BFG](https://rtyley.github.io/bfg-repo-cleaner/))

Reply:
agreed, I had to put it in the repo to get it on the collaborator's server. not the best but it was the moment's quick fix :)

Reply:
I will add it to the Roadmap